init/tmpfiles.py

```diff
@@ -0,0 +1,606 @@
+
+    boot_only = '!' in typ
+    ignore_error = '-' in typ       # TODO implement
+    remove_mismatched = '=' in typ  # TODO implement
+    typ = typ.strip('!-=')
+
+    action = ACTION_MAP[typ](boot_only, mode, user, group, age, arg)
+    return (path, action)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="systemd-tmpfiles clone")
+    parser.add_argument('--dry-run', action='store_true')
+    parser.add_argument('--create', action='store_true')
+    parser.add_argument('--remove', action='store_true')
+    parser.add_argument('--clean', action='store_true')
+    parser.add_argument('--prefix', default='/')
+    args = parser.parse_args()
+
+    if not any([args.create, args.remove, args.clean]):
+        sys.exit("Must specify one or more of --create, --remove, --clean")
+    if args.remove:
+        sys.exit("--remove is not implemented")
+    if args.clean:
+        sys.exit("--clean is not implemented")
+
+    config_files = list_tmpfiles_configs()
+    for config_file in config_files:
+        for path, actions in read_tmpfiles_config(config_file, args.prefix):
+            print(f'>>> {path}')
+            for action in actions:
+                if args.dry_run:
+                    print(action)
+                else:
+                    if args.create:
+                        action.apply(path)
+
+
+def _parse_mode(s):
+    if s is None:
+        return None
+
+    mask = False
+    if s.startswith('~'):
+        mask = True
+        s = s[1:]
+
+    return (mask, int(s, base=8))
+
+
+def _parse_user(s):
+    if s is None:
+        return None
+    try:
+        return int(s)
+    except ValueError:
+        return pwd.getpwnam(s)[2]
+
+
+def _parse_group(s):
+    if s is None:
+        return None
+    try:
+        return int(s)
+    except ValueError:
+        return grp.getgrnam(s)[2]
+
+
+def _parse_attrs(s):
+    if s is None:
+        raise ValueError("file attributes cannot be empty")
+    if len(s) == 0:
+        raise ValueError("file attributes cannot be empty")
+    if s[0] not in "+-=":
+        s = "+" + s  # default operation is '+'
+
+    # TODO more complete checks?  For now just take it as-is.
+    return s
+
+
+def _parse_acls(s):
+    if s is None:
+        raise ValueError("file attributes cannot be empty")
+    if len(s) == 0:
+        raise ValueError("file attributes cannot be empty")
+
+    # TODO more checks?
+    return s
+
+
+def _parse_major_minor(s):
+    if s is None:
+        raise ValueError("major:minor cannot be empty")
+
+    parts = s.split(':')
+    if len(parts) != 2:
+        raise ValueError("major:minor must have 2 parts")
+
+    try:
+        major = int(parts[0])
+        minor = int(parts[1])
+    except ValueError:
+        raise ValueError("major:minor must be integers")
+
+    return (major, minor)
+
+
+class Action():
+    # function to parse/massage argument value
+    # If should raise ValueError for invalid input.
+    arg_function = staticmethod(lambda x: x)
+
```

Comment:
The order in which actions (for a single path) are applied is important.  For example, file/dir must be created before it can be chmod/chown'd.  That is why the mapping is an `OrderedDict` - that's where the order is taken from.  I prefer to explicitly construct this OrderedDict to ensure order is correct, rather than reading from `globals()`.

There may be some use to expressing the action type (and aliases) on the class itself, but I don't have a use for it right now.